v0.23.0 - Windows service can't start

The .msi package for ver 0.23.0 seems to have an issue when installing the service as the provided --log.format flag is invalid. This is the working service path from the previous version:
`"C:\Program Files\windows_exporter\windows_exporter.exe" --log.format logger:eventlog?name=windows_exporter       --config.file="C:\Program Files\windows_exporter\config.yml"
`

While the new version looks like this, missing ?name=windows_exporter
`"C:\Program Files\windows_exporter\windows_exporter.exe" --log.format eventlog       --log.format logger:eventlog?name=windows_exporter --config.file="C:\Program Files\windows_exporter\config.yml"
`

The installation method (double-click default setup or silent with extra flags) doesn't matter. I tried adding the correct --log.format flag to my EXTRA_FLAGS but that just duplicated it and left the invalid flag. I found no documentation on how to edit that flag.

It appears to be an issue with the flag name: `--log.format` should instead be `--log.file` in order to log to eventlog.

Setting the flag to `--log.file` resolves the installation, but the exporter then experiences a panic. Reproducible with the EXE:

```
PS C:\Users\Ben\Downloads> .\windows_exporter-0.23.0-amd64.exe --log.file eventlog
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x10417de]

goroutine 1 [running]:
github.com/go-kit/log.(*syncWriter).Write(0x1ccab60?, {0xc000082100?, 0x15e88e0?, 0x1?})
        C:/Users/runneradmin/go/pkg/mod/github.com/go-kit/log@v0.2.1/sync.go:69 +0xbe
github.com/go-kit/log.logfmtLogger.Log({{0x1843d80?, 0xc000009470?}}, {0xc000300780, 0xa, 0xa})
        C:/Users/runneradmin/go/pkg/mod/github.com/go-kit/log@v0.2.1/logfmt_logger.go:58 +0x19f
github.com/go-kit/log.(*context).Log(0xc00006f090, {0xc000130500?, 0x6, 0x2?})
        C:/Users/runneradmin/go/pkg/mod/github.com/go-kit/log@v0.2.1/log.go:168 +0x34f
github.com/go-kit/log/level.(*logger).Log(0xc00006a140?, {0xc000130500?, 0x15ef7c0?, 0xc00005e7a0?})
        C:/Users/runneradmin/go/pkg/mod/github.com/go-kit/log@v0.2.1/level/level.go:71 +0xa6
github.com/go-kit/log.(*context).Log(0xc00006e550, {0xc00005e7a0?, 0x2, 0x2?})
        C:/Users/runneradmin/go/pkg/mod/github.com/go-kit/log@v0.2.1/log.go:168 +0x34f
github.com/prometheus-community/windows_exporter/collector.newserviceCollector({0x1843da0?, 0xc00006bd00?})
        D:/a/windows_exporter/windows_exporter/collector/service.go:60 +0x16d
github.com/prometheus-community/windows_exporter/collector.Build({0x172a748, 0x7}, {0x1843da0, 0xc00006bd00})
        D:/a/windows_exporter/windows_exporter/collector/collector.go:90 +0x64
main.loadCollectors({0x172a72d, 0x32}, {0x1843da0, 0xc00006bd00})
        D:/a/windows_exporter/windows_exporter/exporter.go:77 +0xa6
main.main()
        D:/a/windows_exporter/windows_exporter/exporter.go:204 +0x194c
```

@jkroepke would you be able to look into the eventlog panic?

--- a/log/logger.go
+++ b/log/logger.go
@@ -24,6 +24,7 @@ func (f *AllowedFile) String() string {
 
 // Set updates the value of the allowed format.
 func (f *AllowedFile) Set(s string) error {
+	f.s = s
 	switch s {
 	case "stdout":
 		f.w = os.Stdout
@@ -77,6 +78,8 @@ func New(config *Config) (log.Logger, error) {
 			return nil, err
 		}
 		l = eventlog.NewEventLogLogger(w, loggerFunc)
+	} else if config.File.w == nil {
+		panic("logger: file writer is nil")
 	} else {
 		l = loggerFunc(log.NewSyncWriter(config.File.w))
 	}

